code/game/machinery/autolathe.dm

```diff
@@ -135,42 +131,25 @@
 		busy = FALSE
 		return 1
 
-	if(HAS_SECONDARY_FLAG(O, HOLOGRAM))
-		return 1
-
-	var/material_amount = materials.get_item_material_amount(O)
-	if(!material_amount)
-		to_chat(user, "<span class='warning'>This object does not contain sufficient amounts of metal or glass to be accepted by the autolathe.</span>")
-		return 1
-	if(!materials.has_space(material_amount))
-		to_chat(user, "<span class='warning'>The autolathe is full. Please remove metal or glass from the autolathe in order to insert more.</span>")
-		return 1
-	if(!user.temporarilyRemoveItemFromInventory(O))
-		to_chat(user, "<span class='warning'>\The [O] is stuck to you and cannot be placed into the autolathe.</span>")
-		return 1
+	return ..()
 
-	busy = TRUE
-	var/inserted = materials.insert_item(O)
-	if(inserted)
-		if(istype(O, /obj/item/stack))
-			if (O.materials[MAT_METAL])
+/obj/machinery/mecha_part_fabricator/ComponentActivated(datum/component/C)
+	if(istype(C, /datum/component/material_container))
+		var/datum/component/material_container/M = C
+		if(!M.last_insert_success)

```

Comment:
Still should probably be cleared somewhere, components shouldn't really rely on being alone.